R/opts.R

```diff
@@ -0,0 +1,447 @@
+#'     )) %>%
+#'   grand_summary_rows(
+#'     columns = vars(currency),
+#'     fns = list(
+#'       total = ~sum(., na.rm = TRUE)
+#'     )) %>%
+#'   tab_source_note(source_note = "This is a source note.") %>%
+#'   tab_footnote(
+#'     footnote = "This is a footnote.",
+#'     locations = cells_body(columns = 1, rows = 1)
+#'   ) %>%
+#'   tab_header(
+#'     title = "The title of the table",
+#'     subtitle = "The table's subtitle"
+#'   ) %>%
+#'   opt_table_lines()
+#'
+#' @family Table Option Functions
+#' @section Function ID:
+#' 9-5
+#'
+#' @export
+opt_table_lines <- function(data,
+                            extent = c("all", "none", "default")) {
+
+  # Perform input object validation
+  stop_if_not_gt(data = data)
+
+  extent <- match.arg(extent)
+
+  # Normalize `extent` values to property values
+  values_vec <- ifelse(extent == "all", "solid", extent)
+
+  # Get vector of `tab_options()` arg names for all specific `locations`
+  options_vec <- table_line_style_vec()
+
+  if (values_vec %in% c("solid", "none")) {
+    option_value_list <- create_option_value_list(options_vec, values_vec)
+    data <- tab_options_multi(data, option_value_list)
+  } else {
+    option_value_list <- create_default_option_value_list(options_vec)
+    data <- tab_options_multi(data, option_value_list)
+  }
+
+  data
+}
+
+#' Option to wrap an outline around the entire table
+#'
+#' This function puts an outline of consistent `style`, `width`, and `color`
+#' around the entire table. It'll write over any existing outside lines so long
+#' as the `width` is larger that of the existing lines. The default value of
+#' `style` (`"solid"`) will draw a solid outline, whereas a value of `"none"`
+#' will remove any present outline.
+#'
+#' @inheritParams fmt_number
+#' @param style,width,color The style, width, and color properties for the table
+#'   outline. By default, these are `"solid"`, `px(3)` (or, `"3px"`), and
+#'   `"#D3D3D3"`.
+#'
+#' @return An object of class `gt_tbl`.
+#'
+#' @examples
+#' # Use `exibble` to create a gt table with
+#' # a number of table parts added; have an
+#' # outline wrap around the entire table by
+#' # using `opt_table_outline()`
+#' tab_1 <-
+#'   exibble %>%
+#'   gt(rowname_col = "row", groupname_col = "group") %>%
+#'   summary_rows(
+#'     groups = "grp_a",
+#'     columns = vars(num, currency),
+#'     fns = list(
+#'       min = ~min(., na.rm = TRUE),
+#'       max = ~max(., na.rm = TRUE)
+#'     )) %>%
+#'   grand_summary_rows(
+#'     columns = vars(currency),
+#'     fns = list(
+#'       total = ~sum(., na.rm = TRUE)
+#'     )) %>%
+#'   tab_source_note(source_note = "This is a source note.") %>%
+#'   tab_footnote(
+#'     footnote = "This is a footnote.",
+#'     locations = cells_body(columns = 1, rows = 1)
+#'   ) %>%
+#'   tab_header(
+#'     title = "The title of the table",
+#'     subtitle = "The table's subtitle"
+#'   ) %>%
+#'   opt_table_outline()
+#'
+#' @section Figures:
```

Comment:
Maybe an example of `style="none"` as well

Reply:
Example added. Thanks.